source/extensions/filters/http/cors/cors_filter.cc

```diff
@@ -115,35 +115,19 @@ void CorsFilter::setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbacks& c
 }
 
 bool CorsFilter::isOriginAllowed(const Http::HeaderString& origin) {
-  return isOriginAllowedString(origin) || isOriginAllowedRegex(origin);
-}
-
-bool CorsFilter::isOriginAllowedString(const Http::HeaderString& origin) {
-  if (allowOrigins() == nullptr) {
-    return false;
-  }
-  for (const auto& o : *allowOrigins()) {
-    if (o == "*" || origin == o.c_str()) {
-      return true;
-    }
-  }
-  return false;
-}
-
-bool CorsFilter::isOriginAllowedRegex(const Http::HeaderString& origin) {
-  if (allowOriginRegexes() == nullptr) {
+  const auto allow_origins = allowOrigins();
+  if (allow_origins == nullptr) {
     return false;
   }
-  for (const auto& regex : *allowOriginRegexes()) {
-    const absl::string_view origin_view = origin.getStringView();
-    if (std::regex_match(origin_view.begin(), origin_view.end(), regex)) {
+  for (const auto& allow_origin : *allow_origins) {
+    if (allow_origin->match("*") || allow_origin->match(origin.getStringView())) {
```

Comment:
@dschaller can you PTAL at all of the CORS changes? I *think* this is correct, but would appreciate you reviewing the changes also. Thank you! cc @htuch

Reply:
This looks good to me. Spinning up the sandbox just to be 100% sure.

If the matcher is an exact string of `*` or a regex pattern of `*` it should match in both cases.

Reply:
Talked to @dschaller offline and he is having trouble with his test env, but is OK with shipping with just a review. @htuch do you mind approving?